cdap-ui/app/cdap/components/DataPrep/DataPrepVisualization/index.js

```diff
@@ -106,7 +106,7 @@ export default class DataPrepVisualization extends Component {
       let {dataprep} = DataPrepStore.getState();
       let properties = dataprep.properties || {};
       let visualization = properties.visualization || {};
-      if (!localData || !this.isArrayEqual(dataprep.data, localData)) {
+      if (!localData || dataprep.data.length !== localData.length || !this.isArrayEqual(localData, dataprep.data)) {
```

Comment:
Isn't the newly added condition already covered by `!this.isArrayEqual...`?

Reply:
no. The isEqual finds the difference between first and second array and does an equality comparison.